### ISSUE TITLE
[Text] Incorrect computation of React.Text height when adding padding

### ISSUE THREAD

**Issue by quentinadam** [facebook/react-native#2466]
When using the React.Text element with padding, the height of text element is incorrectly computed. The height computed is as if the padding had been counted twice, which usually results in a higher text element. This can appear more often with longer text. Here is an example : 

```javascript
'use strict';

var React = require('react-native');
var {
    AppRegistry,
    StyleSheet,
    Text,
    View,
} = React;

var TEXT = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, " +
    "sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. " +
    "Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris " +
    "nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in " +
    "reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla " +
    "pariatur. Excepteur sint occaecat cupidatat non proident, sunt in " +
    "culpa qui officia deserunt mollit anim id est laborum.";

var Test = React.createClass({
    render: function() {
        return (
            <View>
                <Text style={{marginTop: 20, padding: 20, backgroundColor: '#f00'}}>
                    {TEXT}
                </Text>
                <View style={{marginTop: 20, padding: 20, backgroundColor: '#00f'}}>
                    <Text style={{backgroundColor: '#77f'}}>
                        {TEXT}
                    </Text>
                </View>
            </View>
        );
    }
});

AppRegistry.registerComponent('Test', () => Test);
```

Clearly, you would expect the height of the red and the blue boxes to be the same, however this is not the case (see below). This is due to the fact that React seems to compute the height of the text as if the padding on the left and right sides had been deducted twice.

![screen shot 2015-08-27 at 18 07 17](https://cloud.githubusercontent.com/assets/2349972/9525899/b3c53384-4ce6-11e5-9f23-131e41164357.png)

Digging into the React source code, it seems that the problem lies between the `layoutNodeImpl` method of **Layout.c** and the `RCTMeasure` method of **RCTShadowText.m**.
In method `layoutNodeImpl`, the left and right padding are subtracted from the `width` variable. This `width` variable is then passed as argument to the `RCTMeasure` method, where the left and right padding are subtracted again.

**Comment by brentvatne** [comment]
I've noticed this too! It can be very tricky. cc @nicklockwood

**Comment by aleclarson** [comment]
:+1: 

A workaround is to use `margin` instead.

**Comment by quentinadam** [comment]
Indeed, even though if you set a background color and want the text being padded from the background, you need to use an extra parent view (with either margin on the text or padding on the view).
I have submitted a pull request though to correct the behaviour (https://github.com/facebook/react-native/pull/2838)

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/text-incorrect-computation-of-reacttext-height-when-adding-padding

ProductPains helps the community prioritize the most important issues thanks to its voting feature.
It is easy to use - just login with GitHub.

Also, if this issue is a bug, please consider sending a PR with a fix. We rely on the community to provide
bugfixes as the core team is focused on performance.

**Comment by aleclarson** [comment]
@mkonicek I think this issue has been fixed in master already? c73ceff
